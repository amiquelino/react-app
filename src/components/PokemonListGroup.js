import React from 'react'
import {ListGroup, ListGroupItem} from 'react-bootstrap'
const PokemonListGroup = ( props ) => {

  return (
    <ListGroup >
      {props.dataset.map((pokemon) => {
        return (
          <ListGroupItem href={pokemon.url} key={pokemon.name}> {pokemon.name} </ListGroupItem>
        )
      })}
    </ListGroup>
  )
}

PokemonListGroup.propTypes = {
  dataset: React.PropTypes.array.isRequired
}

export default PokemonListGroup
