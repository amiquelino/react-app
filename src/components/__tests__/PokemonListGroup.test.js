import React from 'react'
import renderer from 'react-test-renderer'
import PokemonListGroup from '../PokemonListGroup'

describe('PokemonListGroup', ( ) => {
  const pokemons = [
    {
      "url" : "http://pokeapi.co/api/v2/pokemon/1/",
      "name":"bulbasaur"
    },
    {
      "url" : "http://pokeapi.co/api/v2/pokemon/2/",
      "name" : "ivysaur"
    }
  ]

  it('render correctly', ( ) => {
    const component = renderer.create(
      <PokemonListGroup
        dataset={pokemons}
        next={'http://pokeapi.co/api/v2/pokemon/?offset=20'}
        previous={null}
        previousOnClick={jest.fn()}
        nextOnClick={jest.fn()}
      />
    )
    expect( component ).toMatchSnapshot( )
  })
})
