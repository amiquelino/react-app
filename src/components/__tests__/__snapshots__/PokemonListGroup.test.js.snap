exports[`PokemonListGroup render correctly 1`] = `
<div
  className="list-group">
  <a
    className="list-group-item"
    href="http://pokeapi.co/api/v2/pokemon/1/">
     
    bulbasaur
     
  </a>
  <a
    className="list-group-item"
    href="http://pokeapi.co/api/v2/pokemon/2/">
     
    ivysaur
     
  </a>
</div>
`;
